File: Exercise_50.py
# Create a function that will receive n as argument and return an array of n
# random numbers from 1 to n. The numbers should be unique inside the array.
import random
def array_of_random_numbers(n):
    result = []
    count = 0
    while count < n:
        x = random.randint(1, n)
        if x not in result:
            result.append(x)
            count += 1
    return result
print(array_of_random_numbers(10))

File: Randy-JRA/Exercise_50.cpp
//Challenge 50 is to Create a function that will receive n as argument and return an array of n
//random numbers from 1 to n

#include <iostream>
#include <cstdlib>
#include <ctime>

int* generateRandomArray(int n) {
    int* arr = new int[n]; // allocate memory for the array
    srand(time(NULL)); // seed the random number generator with current time
    for (int i = 0; i < n; i++) {
        arr[i] = rand() % n + 1; // generate a random number between 1 and n
    }
    return arr; // return the array
}

int main() {
    int n = 10; // example usage
    int* arr = generateRandomArray(n);
    for (int i = 0; i < n; i++) {
        std::cout << arr[i] << " ";
    }
    delete[] arr; // free the memory allocated for the array
    return 0;
}
 

